### PR TITLE
add data_provider

### PR DESCRIPTION
**Pull request by Resistendi**

### CODE REVIEW

File: transport/data_provider.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        self.dbx = dropbox.Dropbox(access_token)
+
+    def __get_newest_monefy_data_name(self, list_of_files):
+        newest_file_index = -1
+        newest_file_date = datetime(1, 1, 1)
+        for i in range(len(list_of_files)):
+            file_name_parts = list_of_files[i].split('.')
+            if (file_name_parts[0] == 'Monefy' and
+                    file_name_parts[1] == 'Data'):
+                if datetime.strptime(
```
Comment (IrinaLokhvytska) [inline transport/data_provider.py]: file_name_parts contains values like this `['Monefy', 'Data', '28', '03', '18', 'csv']`, so this part `datetime.strptime(file_name_parts[2], "%d-%m-%y")` causes an error

File: transport/data_provider.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                        file_name_parts[2], "%m-%d-%y") > newest_file_date:
```
Comment (IrinaLokhvytska) [inline transport/data_provider.py]: `f_date = '-'.join([file_name_parts[2], file_name_parts[3], file_name_parts[4]])
                if datetime.strptime(f_date, "%d-%m-%y") > newest_file_date:
                    newest_file_date = datetime.strptime(f_date, "%d-%m-%y")
                    newest_file_index = i`

File: transport/data_provider.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    newest_file_date == datetime.strptime(
```
Comment (IrinaLokhvytska) [inline transport/data_provider.py]: Change == to =

File: transport/data_provider.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        Warning: 'download_path' are relative to module where its called.
+
+        :param download_path: path to download (should include file name).
+        :param file_name: name of file to download.
+        :return:
+        '''
+        self.dbx.files_download_to_file(download_path, '/' + file_name)
+
+    def get_newest_monefy_data(self, download_path):
```
Comment (IrinaLokhvytska) [inline transport/data_provider.py]: I think, we need to specify the path in the constructor. 
self.download_path = <root directory>/download

File: transport/data_provider.py (hunk 1 above)
Comment (Resistendi) [reply transport/data_provider.py]: file name looks like 'Monefy.Data.4-5-18.csv' i've tested it manually, works correct

File: transport/data_provider.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    This is DataProvider class
+    Methods:
+        get_files_list
+        get_file_by_name
+        get_newest_monefy_data
+    '''
+    def __init__(self, access_token):
+        self.dbx = dropbox.Dropbox(access_token)
+
+    def __get_newest_monefy_data_name(self, list_of_files):
```
Comment (evgeniyDorogoy) [inline transport/data_provider.py]: Lets refactor this method using http://dropbox-sdk-python.readthedocs.io/en/latest/moduledoc.html?highlight=sort#dropbox.files.CommitInfo. Idea is in creating list of `file_path` - `client_modified` tuples and sort it by `client_modified`.

**Comment by evgeniyDorogoy** [comment]
Don't forget to update `CHANGELOG.md`

File: transport/data_provider.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        '''
+        if self._mode == 'manual':
+            newest_file_name = \
+                self.__get_newest_monefy_data_name_manual(
+                    self.get_files_list())
+        elif self._mode == 'auto':
+            newest_file_name = \
+                self.__get_newest_monefy_data_name_auto()
+
+        if newest_file_name:
```
Comment (IrinaLokhvytska) [inline transport/data_provider.py]: Declare a variable before using it:
newest_file_name = ''

File: transport/data_provider.py [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+            directory = ''
+
+        list_of_files = self.dbx.files_list_folder(directory).entries
+        list_of_files.reverse()
+
+        for entry in list_of_files:
+            if entry.name.startswith('Monefy.Data'):
+                return entry.name
+
+        return None
```
Comment (IrinaLokhvytska) [inline transport/data_provider.py]: A condition without return returns None. Probably it is not necessary to write  `return None`, at you above specified, that function should return a string.

**Pull request merged**